modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -809,6 +834,22 @@ private void validateWorkerActionParameters(BLangWorkerSend send, BLangWorkerRec
         }
     }
 
+    private boolean checkNextBreakValidityInTransaction() {
+        boolean invalid = false;
```

Comment:
Can we simplify the logic to "return !loopWithintransactionCheckStack.peek()". The stack will always have elements, since this check method is called after validating if the next/continue is inside a loop.

Reply:
Fixed with f6fe91e29e529078b0c173726ea698ee63e9fe62. Thanks